atom/utility/atom_content_utility_client.cc

```diff
@@ -4,30 +4,74 @@
 
 #include "atom/utility/atom_content_utility_client.h"
 
+#include <utility>
+
+#include "base/command_line.h"
 #include "content/public/child/child_thread.h"
+#include "content/public/common/service_manager_connection.h"
+#include "content/public/common/simple_connection_filter.h"
 #include "services/proxy_resolver/proxy_resolver_service.h"
 #include "services/proxy_resolver/public/mojom/proxy_resolver.mojom.h"
+#include "services/service_manager/sandbox/switches.h"
+
+#if BUILDFLAG(ENABLE_PRINTING_ELECTRON)
+#include "chrome/services/printing/printing_service.h"
+#include "chrome/services/printing/public/mojom/constants.mojom.h"
 
 #if defined(OS_WIN)
-#include "chrome/utility/printing_handler_win.h"
-#endif
+#include "chrome/services/printing/pdf_to_emf_converter_factory.h"
+#include "chrome/utility/printing_handler.h"
+#endif  // defined(OS_WIN)
+
+#endif  // BUILDFLAG(ENABLE_PRINTING_ELECTRON)
 
 namespace atom {
 
-AtomContentUtilityClient::AtomContentUtilityClient() {
-#if defined(OS_WIN)
-  handlers_.push_back(std::make_unique<printing::PrintingHandlerWin>());
+AtomContentUtilityClient::AtomContentUtilityClient() : elevated_(false) {
+#if BUILDFLAG(ENABLE_PRINTING_ELECTRON) && defined(OS_WIN)
+  printing_handler_ = std::make_unique<printing::PrintingHandler>();
 #endif
 }
 
 AtomContentUtilityClient::~AtomContentUtilityClient() {}
 
-bool AtomContentUtilityClient::OnMessageReceived(const IPC::Message& message) {
-#if defined(OS_WIN)
-  for (const auto& handler : handlers_) {
-    if (handler->OnMessageReceived(message))
-      return true;
+void AtomContentUtilityClient::UtilityThreadStarted() {
+#if BUILDFLAG(ENABLE_PRINTING_ELECTRON) && defined(OS_WIN)
+  base::CommandLine* command_line = base::CommandLine::ForCurrentProcess();
+  elevated_ = command_line->HasSwitch(
+      service_manager::switches::kNoSandboxAndElevatedPrivileges);
+#endif
+
+  content::ServiceManagerConnection* connection =
+      content::ChildThread::Get()->GetServiceManagerConnection();
+
+  // NOTE: Some utility process instances are not connected to the Service
```

Comment:
... some? which ones? why?

Reply:
Weeeell, I got this from the chromium equivalent, I think this part could be dropped actually, but I didn't want to just delete something from there since I don't really know what it would break. So I'm not sure.

Reply:
Ah, would be great to point at the chromium equivalent in a comment!